### PR TITLE
[aptos-tool] Add testing and remove key generator crate

### PR DESCRIPTION
**Pull request by gregnazario**
## Motivation

Remove the key generator crate and setup framework for testing with new op tool

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

/canary

### CODE REVIEW

File: crates/aptos-faucet/Cargo.toml [hunk 1]
```diff
@@ -24,7 +24,7 @@ tokio = { version = "1.8.1", features = ["full"] }
 url = "2.2.2"
 warp = "0.3.2"
 
-generate-key = { path = "../../config/generate-key" }
+aptos = { path = "../aptos" }
```
Comment (JoshLind) [inline crates/aptos-faucet/Cargo.toml:27]: A bit late to the party but naming the crate `aptos` seems a bit vague to me. If this is tooling, while not `aptos-tool` or `aptos-cli`? Or is this also going to include compilable library code, e.g., to aid Rust development for Aptos services and apps?

File: crates/aptos/src/op/key.rs [hunk 2]
```diff
@@ -62,6 +64,51 @@ impl GenerateKey {
...
+        command.execute()?;
+        Ok((
+            load_key(key_file, encoding_type)?,
+            load_key(&append_file_extension(key_file, ".pub")?, encoding_type)?,
+        ))
+    }
+
+    /// Generates an `Ed25519PrivateKey` without saving it to disk
+    pub fn generate_ed25519_locally() -> ed25519::Ed25519PrivateKey {
+        // Generate a ed25519 key
```
Comment (JoshLind) [inline crates/aptos/src/op/key.rs:108]: nit: remove this comment?

File: crates/aptos/src/op/key.rs [hunk 3]
```diff
@@ -62,6 +64,51 @@ impl GenerateKey {
...
+        let command = GenerateKey::from_iter(args.split_whitespace());
+        command.execute()?;
+        Ok((
+            load_key(key_file, encoding_type)?,
+            load_key(&append_file_extension(key_file, ".pub")?, encoding_type)?,
+        ))
+    }
+
+    /// Generates an `Ed25519PrivateKey` without saving it to disk
+    pub fn generate_ed25519_locally() -> ed25519::Ed25519PrivateKey {
```
Comment (JoshLind) [inline crates/aptos/src/op/key.rs:107]: nit: maybe `new_in_memory_ed25519` or `generate_ed25519_in_mem`? `locally` made me think we're using the local random number generator (e.g., OS) and not that we weren't writing it to a file.

File: crates/aptos/src/op/key.rs [hunk 4]
```diff
@@ -62,6 +64,51 @@ impl GenerateKey {
...
+        let command = GenerateKey::from_iter(args.split_whitespace());
+        command.execute()?;
+        Ok((
+            load_key(key_file, encoding_type)?,
+            load_key(&append_file_extension(key_file, ".pub")?, encoding_type)?,
+        ))
+    }
+
+    /// A test friendly typed key generation for e25519 keys.
+    pub fn generate_ed25519(
```
Comment (JoshLind) [inline crates/aptos/src/op/key.rs:88]: nit: Rename to `generate_ed25519_on_disk`? If this naming convention makes sense, we could also adopt it for `x25519`.

File: crates/aptos/src/op/key.rs [hunk 5]
```diff
@@ -62,6 +64,51 @@ impl GenerateKey {
...
+            "generate --key-type {key_type:?} --key-file {key_file} --encoding {encoding_type:?} --assume-yes",
+            key_type = KeyType::X25519,
+            key_file = key_file.to_str().unwrap(),
+            encoding_type = encoding_type,
+        );
+        let command = GenerateKey::from_iter(args.split_whitespace());
+        command.execute()?;
+        Ok((
+            load_key(key_file, encoding_type)?,
+            load_key(&append_file_extension(key_file, ".pub")?, encoding_type)?,
```
Comment (JoshLind) [inline crates/aptos/src/op/key.rs:83]: nit: I'd maybe move `.pub` to a global and reference it multiple times (so it's obvious what file conventions we're using)

File: crates/aptos/src/op/key.rs [hunk 6]
```diff
@@ -150,3 +190,31 @@ fn check_if_file_exists(file: &Path, assume_yes: bool) -> Result<(), Error> {
         Ok(())
     }
 }
+
+/// Loads a key to a file hex string encoded
```
Comment (JoshLind) [inline crates/aptos/src/op/key.rs:194]: Should this read `Loads a key from a file`?

File: testsuite/smoke-test/src/operational_tooling.rs [hunk 7]
```diff
@@ -606,47 +599,65 @@ async fn test_extract_peers_from_keys() {
...
-    assert_eq!(
-        expected_key,
-        x25519::PrivateKey::try_from(
-            base64::decode(fs::read(path.as_ref()).unwrap())
-                .unwrap()
-                .as_slice()
-        )
-        .unwrap(),
-    );
+    let (priv_key, pub_key) =
```
Comment (JoshLind) [inline testsuite/smoke-test/src/operational_tooling.rs:607]: The pedantic person inside of me would make a separate unit test for each of the key encodings we're testing in here 😄

**Review by JoshLind — COMMENTED** [review]
Thanks @gregnazario. Bunch of nits -- nothing serious. Let me know what you think. 😄

File: crates/aptos-faucet/Cargo.toml (hunk 1 above)
Comment (gregnazario) [reply crates/aptos-faucet/Cargo.toml:27]: We can change this later.

File: testsuite/smoke-test/src/operational_tooling.rs (hunk 7 above)
Comment (gregnazario) [reply testsuite/smoke-test/src/operational_tooling.rs:607]: Yeah, I wasn't going to rewrite it for multiple tests though :P

**Review by JoshLind — APPROVED** [review]
LGTM! 😄

**Pull request closed without merging**